File: 2026scon/problems/where-is-my-icp/solutions/chleee_ac.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, m;
    string s, t;
    cin >> n >> s >> m >> t;
    string target = "ICPC";
    map<char,int> need;
    for (char c : target) need[c]++;
    for (char c : s) need[c]--;
    for (char c : t) need[c]--;
    bool ok = true;
    for (auto &[k,v] : need)
        if (v > 0) ok = false;
    cout << (ok ? "m4us happy" : "m4us sad") << endl;
}

File: 2026scon/problems/where-is-my-icp/solutions/chleee_ac.py
from collections import Counter
N = int(input())
S = input()
M = int(input())
T = input()
need = Counter("ICPC")
need.subtract(Counter(S))
have = Counter(T)
for c in need:
    if need[c] > have[c]:
        print("m4us sad")
        break
else:
    print("m4us happy")
